bundles/org.openhab.binding.sedif/src/main/java/org/openhab/binding/sedif/internal/discovery/SedifDiscoveryService.java

```diff
@@ -0,0 +1,146 @@
+import org.openhab.binding.sedif.internal.dto.Contract;
+import org.openhab.binding.sedif.internal.dto.ContractDetail;
+import org.openhab.binding.sedif.internal.dto.ContractDetail.CompteInfo;
+import org.openhab.binding.sedif.internal.handler.BridgeSedifWebHandler;
+import org.openhab.binding.sedif.internal.helpers.SedifListener;
+import org.openhab.binding.sedif.internal.types.SedifException;
+import org.openhab.core.config.discovery.AbstractThingHandlerDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ServiceScope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SedifDiscoveryService} class is the service to discover a skeleton for controller handlers.
+ *
+ * @author Laurent Arnal - Initial contribution
+ */
+@Component(scope = ServiceScope.PROTOTYPE, service = SedifDiscoveryService.class)
+@NonNullByDefault
+public class SedifDiscoveryService extends AbstractThingHandlerDiscoveryService<BridgeSedifWebHandler>
+        implements SedifListener {
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Set.of(THING_TYPE_METER);
+    private static final int SCAN_DURATION_IN_S = 60;
+
+    private final Logger logger = LoggerFactory.getLogger(SedifDiscoveryService.class);
+
+    public SedifDiscoveryService() {
+        super(BridgeSedifWebHandler.class, SCAN_DURATION_IN_S);
+    }
+
+    public SedifDiscoveryService(BridgeSedifWebHandler controllerHandler) {
+        this();
+        setThingHandler(controllerHandler);
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SUPPORTED_THING_TYPES;
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.debug("Start Sedif device background discovery");
+        thingHandler.addListener(this);
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        logger.debug("Stop Sedif device background discovery");
+        thingHandler.removeListener(this);
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Sedif discovery: Start {}", thingHandler.getThing().getUID());
+
+        // Start the search for new devices
+        thingHandler.addListener(this);
+    }
+
+    @Override
+    public synchronized void abortScan() {
+        logger.debug("Sedif discovery: Abort {}", thingHandler.getThing().getUID());
+        thingHandler.removeListener(this);
+        super.abortScan();
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        logger.debug("Sedif discovery: Stop {}", thingHandler.getThing().getUID());
+        thingHandler.removeListener(this);
+        super.stopScan();
```

Comment:
I believe you don't need to add/remove bridge listener each time a manual discovery is run.
This is certainly something to be done globally when the discovery service is activated/deactivated.